vmicore/src/os/windows/KernelAccess.cpp

```diff
@@ -168,23 +168,60 @@ uint32_t KernelAccess::extractFlagValue32(addr_t flagBaseVA, size_t startBit, si
     return getFlagValue(flagValue, startBit, endBit);
 }
 
+uint64_t KernelAccess::extractFlagValue64(addr_t flagBaseVA, size_t startBit, size_t endBit) const
+{
+    expectSaneKernelAddress(flagBaseVA, __func__); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    auto flagValue = vmiInterface->read64VA(flagBaseVA, vmiInterface->getSystemCr3());
+
+    return getFlagValue(flagValue, startBit, endBit);
+}
+
 addr_t KernelAccess::getMmVadShortFlagsAddr(addr_t vadShortBaseVA) const
 {
     return vadShortBaseVA + kernelOffsets.mmVadShort.Flags;
 }
 
 uint32_t KernelAccess::extractProtectionFlagValue(addr_t vadShortBaseVA) const
 {
-    return extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
-                              kernelOffsets.mmvadFlags.protection.startBit,
-                              kernelOffsets.mmvadFlags.protection.endBit);
+    auto flagStructSize = vmiInterface->getStructSizeFromJson(kernelOffsets.mmvadFlags.structName);
+    switch (flagStructSize)
+    {
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+        case 4:
+            return extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
+                                      kernelOffsets.mmvadFlags.protection.startBit,
+                                      kernelOffsets.mmvadFlags.protection.endBit);
+        // As of now, there are 32 Protectionvalues
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+        case 8:
+            return static_cast<uint32_t>(extractFlagValue64(getMmVadShortFlagsAddr(vadShortBaseVA),
+                                                            kernelOffsets.mmvadFlags.protection.startBit,
+                                                            kernelOffsets.mmvadFlags.protection.endBit));
+        default:
+            throw VmiException(std::string(kernelOffsets.mmvadFlags.structName) + ": " +
+                               Convenience::intToDec(flagStructSize) + " is unknown flag struct size");
+    }
 }
 
 bool KernelAccess::extractIsPrivateMemory(addr_t vadShortBaseVA) const
 {
-    return static_cast<bool>(extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
-                                                kernelOffsets.mmvadFlags.privateMemory.startBit,
-                                                kernelOffsets.mmvadFlags.privateMemory.endBit));
+    auto flagStructSize = vmiInterface->getStructSizeFromJson(kernelOffsets.mmvadFlags.structName);
+    switch (flagStructSize)
+    {
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+        case 4:
+            return static_cast<bool>(extractFlagValue32(getMmVadShortFlagsAddr(vadShortBaseVA),
```

Comment:
nit: Same here. Add an assert statement in order to verify that this is actually a single flag.